src/visitors/Tag.js

```diff
@@ -57,8 +67,8 @@ function getAttributes(
 
       const expr = parseExpression(val === true ? 'true' : val, context);
 
-      if (!mustEscape && (!t.isStringLiteral(expr) || /(\<\>\&)/.test(val))) {
```

Comment:
Why are you removing the `(!t.isStringLiteral(expr) || /(\<\>\&)/.test(val))`?

Reply:
@ForbesLindesay because it looks impossible to achieve with basic tests. Maybe we can hack it somehow... Otherwise `val` starts and ends with `"` when it's a string. 

And filtering off `<>&` looks not obvious for me, why not `<script>...` or something like that? Probably we lost `g` flag for regex?

Anyway I'm going to use different conditions. All use cases are captured in tests. Let's discuss something, if you want to add to those tests.

Reply:
The case this was capturing, although arguably not very important, is:

```pug
div(data-whatever!="some string that does not contain any special characters")
```

The idea being that it might make it slightly easier when converting from html based pug templates to react pug templates.  Arguably that might not be super important though.